unumpy/linalg/_multimethods.py

```diff
@@ -0,0 +1,198 @@
+import functools
+import operator
+from uarray import create_multimethod, mark_as, all_of_type, Dispatchable
+import builtins
+
+create_numpy = functools.partial(create_multimethod, domain="numpy.linalg")
+
+from .._multimethods import (
+    ndarray,
+    _identity_argreplacer,
+    _self_argreplacer,
+    _dtype_argreplacer,
+    mark_dtype,
+    _first_argreplacer,
+    _first2argreplacer,
+)
+
+
+@create_numpy(_first_argreplacer)
+@all_of_type(ndarray)
+def multi_dot(arrays):
+    return arrays
+
+
+def matrix_power_default(a, n):
+    if a.shape[0] == a.shape[1]:
```

Comment:
This needs to be re-written in terms of the eigenvalue decomposition. If you're unsure what that is, here's a Wikipedia page (in NumPy it's the `eig` function): https://en.wikipedia.org/wiki/Eigendecomposition_of_a_matrix

https://math.stackexchange.com/a/2552148/59208

Reply:
You are right, I misunderstood what matrix_power does when I wrote that.